src/Microsoft.AspNetCore.Routing/Patterns/RoutePatternPart.cs

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Routing.Patterns
+{
+    public abstract class RoutePatternPart
+    {
+        // This class is **not** an extensibility point - every part of the routing system
+        // needs to be aware of what kind of parts we support.
+        //
+        // It is abstract so we can add semantics later inside the library.
+        internal RoutePatternPart(RoutePatternPartKind partKind)
```

Comment:
Did the private AND internal keyword ever get added to C#? A candidate to use it right here 😝

Reply:
C# 7.2! DOPE